### PR TITLE
Minor RocksJava Java code cosmetics

### PR DESCRIPTION
**Pull request by stefan-zobel**
Specifically:
- unused imports
- code formatting
- typos in comments
- unnecessary casts
- missing default label in switch statement
- explicit use of long literals in multiplication
- use generics where possible without backward compatibility risk

### CODE REVIEW

**Comment by stefan-zobel** [comment]
Sorry, I really don't get what the format check is complaining about.

**Comment by adamretter** [comment]
@stefan-zobel You can just run `make format` and it will prompt you to reformat any issues for you and allow you to append a commit

File: java/src/main/java/org/rocksdb/MutableColumnFamilyOptionsInterface.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
    * @throws java.lang.IllegalArgumentException thrown on 32-Bit platforms
    *   while overflowing the underlying platform specific value.
    */
-  MutableColumnFamilyOptionsInterface setWriteBufferSize(long writeBufferSize);
+  MutableColumnFamilyOptionsInterface<T> setWriteBufferSize(long writeBufferSize);
```
Comment (adamretter) [inline java/src/main/java/org/rocksdb/MutableColumnFamilyOptionsInterface.java:29]: Why not just `T` as per the other functions?

Comment (stefan-zobel) [reply java/src/main/java/org/rocksdb/MutableColumnFamilyOptionsInterface.java:29]: @adamretter That sounds sensible. It just didn't occur to me.

**Comment by stefan-zobel** [comment]
> @stefan-zobel You can just run `make format` and it will prompt you to reformat any issues for you and allow you to append a commit

```
build_tools/format-diff.sh
Checking format of uncommitted changes...
Nothing needs to be reformatted!
```

**Comment by stefan-zobel** [comment]
I'll try again ...

**Comment by stefan-zobel** [comment]
To no avail. Still format error and `make format` says nothing needs to be reformatted.
I' withdrawing this PR.

**Comment by adamretter** [comment]
@stefan-zobel Could it be that you make an older version of the tools used by make format installed on your machine? The output from `make format` in CI is fairly clear - https://github.com/facebook/rocksdb/runs/4305435864?check_suite_focus=true#step:10:1

**Comment by stefan-zobel** [comment]
@adamretter
I was running it on Manajaro and used clang-format-static-bin from the Arch User Repository. That should be very current.
My guess the problem is that it is only looking at the uncomitted changes as its output says. I had to manually edit each already committed file  to make it do anything.

**Comment by stefan-zobel** [comment]
Reopening on more time

**Comment by stefan-zobel** [comment]
> That hasn't been my experience, my experience is that it considers all my commits in a branch and offers to either amend the formats to the last commit or leave them staged so that I can create just a formatting commit

@adamretter
Interesting. Yes, it's sort of weird.

And it gets even more confusing when the format check nags about formatting errors in parts of the code where the formatting hasn't changed at all.
For example, I didn't touch the formatting of the comment in `CompactRangeOptions` (my only change was to replace the 4 letters "neum" by the 4 letters "enum") and something similar holds for the format of the `TransactionalDB` interface declaration.
It took me a while to realize that the check wants me to change the formatting anyway.

**Comment by adamretter** [comment]
> I didn't touch the formatting of the comment

Perhaps it is the case that somehow that code was committed before the formatting checks were introduced, now that you touch that area of the code the formatted complains within the context of your non-format related change?

**Comment by stefan-zobel** [comment]
> > I didn't touch the formatting of the comment
> 
> Perhaps it is the case that somehow that code was committed before the formatting checks were introduced, now that you touch that area of the code the formatter complains within the context of your non-format related change?

Sure. There is no other sensible explanation. By chance I've noticed one other place that wouldn't survive the format check if someone would change a single letter.

**Comment by adamretter** [comment]
@stefan-zobel Did you mean to close this?

**Comment by stefan-zobel** [comment]
> @stefan-zobel Did you mean to close this?

@adamretter

Yes, why leave it open if no one seems to be interested?

**Comment by adamretter** [comment]
@stefan-zobel I don't think it is that no one is interested. In the RocksDB project it can take some time to get code merged. Let me see... @jay-zhuang @mrambacher Can you get this one merged please?

**Review by jay-zhuang — APPROVED** [review]
LGTM

**Pull request closed without merging**